tests/test_identity_pool.py

```diff
@@ -551,3 +564,177 @@ def test_refresh_with_retrieve_subject_token_error(self):
                 SUBJECT_TOKEN_JSON_FILE, "not_found"
             )
         )
+
+    def test_retrieve_subject_token_from_url(self):
+        credential_source = {
+            "url": "http://fakeurl.com",
+        }
+        credentials = self.make_credentials(credential_source=credential_source)
+        subject_token = credentials.retrieve_subject_token(
+            self.make_mock_request(token_data=TEXT_FILE_SUBJECT_TOKEN))
+
+        assert subject_token == TEXT_FILE_SUBJECT_TOKEN
+
+    def test_retrieve_subject_token_from_url_json(self):
+        credential_source = {
+            "url": "http://fakeurl.com",
+            "format": {"type": "json", "subject_token_field_name": "access_token"},
+        }
+        credentials = self.make_credentials(credential_source=credential_source)
+        subject_token = credentials.retrieve_subject_token(
+            self.make_mock_request(token_data=JSON_FILE_CONTENT))
+
+        assert subject_token == JSON_FILE_SUBJECT_TOKEN
```

Comment:
Can you also assert the request was sent with the expected parameters? eg. the expected URL, GET method and headers and no body? We don't have any coverage for this at the moment.